cibuildwheel/macos.py

```diff
@@ -271,22 +271,29 @@ def setup_python(
     # PyPy defaults to 10.7, causing inconsistencies if it's left unset.
     env.setdefault("MACOSX_DEPLOYMENT_TARGET", "10.9")
 
+    def macos_depl(sys_value: str) -> str:
+        "Return the current target, or the system minimum, whichever is higher"
+        cur_value = env["MACOSX_DEPLOYMENT_TARGET"]
+        cur_value_tup = tuple(int(v) for v in cur_value.split("."))
+        sys_value_tup = tuple(int(v) for v in sys_value.split("."))
+        return cur_value if cur_value_tup > sys_value_tup else sys_value
```

Comment:
Should we force `{major}.{minor}` to be returned ? I think so.
If major >= 11, we should force minor to 0 given the new versioning scheme of macOS

Reply:
Yes, you are correct. I thought we did this, I'm remembering the "force to 0 part" - must have been in scikit-build or elsewhere.